tools/plot-bench.py

```diff
@@ -0,0 +1,305 @@
+    parser.add_argument('--filetype', default='png',  metavar='BACKEND',
+        help='the file type used, these can be found on the matplotlib website.')
+    parser.add_argument('--quality', default='low', metavar='QUALITY',
+        choices=['low', 'medium', 'high'], help='The quality of the plot')
+    parser.add_argument('--transparent', action='store_true',
+        help='flag to use if the bagground should be transparent.')
+
+    return parser.parse_args()
+
+
+def format_arg_list(args: Optional[str]) -> Optional[str]:
+    """
+    Takes a string of form 'a, b, c, d' and makes a list ['a', 'b', 'c', 'd']
+    """
+    if args is None:
+        return None
+    
+    return set(map(lambda arg: arg.strip(), args.split(',')))
+
+
+def make_plot_jobs_and_directories(
+    programs: Set[str],
+    plot_jobs: Dict[str, Dict[str, Dict[str, Any]]],
+    data: Dict[str, Dict[str, Dict[str, Any]]],
+    plots: Set[str],
+    filetype: str
+    ) -> None:
+    """Makes dictionary with plot jobs where plot_jobs are the jobs."""
+
+    for program in programs:
+        datasets = data.get(program).get('datasets')
+        for dataset_path, dataset in datasets.items():
+            runtimes = dataset.get('runtimes')
+            directory = os.path.dirname(program)
+            program_basename = pathlib.Path(program).stem
+            dataset_basename = pathlib.Path(dataset_path).stem
+            dataset_name = textwrap.shorten(dataset_basename, 50).replace(' ', '-')
+            os.makedirs(directory, exist_ok=True)
+            for plot in plots:
+                filename = f'{program_basename}-{dataset_name}-{plot}.{filetype}'
+                plot_file = os.path.join(directory, filename)
+                plot_jobs[plot][plot_file] = {
+                    'program': program_basename,
+                    'dataset': dataset_basename,
+                    'runtimes': np.array(runtimes)
+                }
+
+def set_quality(quality: str) -> int:
+    """The quality is set based on the string given and the dpi is returned."""
+
+    if quality == 'low':
+        return 100
+    elif quality == 'medium':
+        return 200
+    elif quality == 'high':
+        plt.rcParams.update({
+            'text.usetex': True,
+            'font.family': 'serif',
+            'font.serif': ['Computer Modern Serif']
+        })
+        return 400
+    
+    raise Exception(f'{quality} is an unknown quality.')
+
+def main():
+    plt.rcParams.update({
+        'ytick.color': 'black',
+        'xtick.color': 'black',
+        'axes.labelcolor': 'black',
+        'axes.edgecolor': 'black',
+        'axes.axisbelow': True,
+        'text.usetex': False,
+        'axes.prop_cycle': matplotlib.cycler(color=['#5f021f'])
+    })
+
+    args = get_args()
+    data = json.load(open(args.filename, 'r'))
+    plots = format_arg_list(args.plots)
+    programs = format_arg_list(args.programs)
+    filetype = args.filetype
+    quality = args.quality
```

Comment:
Realistically, nobody is going to set this.  It is *your* responsibility as the programmer to pick what is right, rather than pushing it on to your users.  Since graphs compress losslessly very well as PNG files, and the sizes don't matter, I would prioritize visual quality.

Reply:
I have removed quality now and now it just uses png dpi=200